packages/core/chrome/core-chrome-browser-internal/src/project_navigation/breadcrumbs.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  AppDeepLinkId,
+  ChromeProjectBreadcrumb,
+  ChromeProjectNavigationNode,
+  ChromeSetProjectBreadcrumbsParams,
+  ChromeBreadcrumb,
+} from '@kbn/core-chrome-browser';
+import { createHomeBreadcrumb } from './home_breadcrumbs';
+
+export function buildBreadcrumbs({
+  homeHref,
+  projectBreadcrumbs,
+  activeNodes,
+  chromeBreadcrumbs,
+}: {
+  homeHref: string;
+  projectBreadcrumbs: {
+    breadcrumbs: ChromeProjectBreadcrumb[];
+    params: ChromeSetProjectBreadcrumbsParams;
+  };
+  chromeBreadcrumbs: ChromeBreadcrumb[];
+  activeNodes: ChromeProjectNavigationNode[][];
+}): ChromeProjectBreadcrumb[] {
+  const homeBreadcrumb = createHomeBreadcrumb({
+    homeHref,
+  });
+
+  if (projectBreadcrumbs.params.absolute) {
+    return [homeBreadcrumb, ...projectBreadcrumbs.breadcrumbs];
+  }
+
+  // breadcrumbs take the first active path
+  const activePath: ChromeProjectNavigationNode[] = activeNodes[0] ?? [];
+  const navBreadcrumbPath = activePath.filter(
+    (n) => Boolean(n.title) && n.breadcrumbStatus !== 'hidden'
+  );
+  const navBreadcrumbs = navBreadcrumbPath.map(
+    (node): ChromeProjectBreadcrumb => ({
+      href: node.deepLink?.url ?? node.href,
+      deepLinkId: node.deepLink?.id as AppDeepLinkId,
```

Comment:
Why is this casting needed? On L63 you are checking if it is not `undefined`. Maybe the `node` interface in the argument needs to be updated?

Reply:
EDIT: Re-reading this, I now see why you had to do it, `id` are `string` in the `DeepLink` interface. 👍